packages/react-core/src/components/DualListSelector/DualListSelectorTreeItem.tsx

```diff
@@ -0,0 +1,154 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/DualListSelector/dual-list-selector';
+import { css } from '@patternfly/react-styles';
+import { DualListSelectorTreeItemData } from './DualListSelectorTree';
+import { Badge } from '../Badge';
+import AngleRightIcon from '@patternfly/react-icons/dist/js/icons/angle-right-icon';
+import { flattenTree } from './treeUtils';
+
+export interface DualListSelectorTreeItemProps extends React.HTMLProps<HTMLLIElement> {
+  /** Content rendered inside the dual list selector. */
+  children?: React.ReactNode;
+  /** Additional classes applied to the dual list selector. */
+  className?: string;
+  /** Flag indicating the option is currently selected. */
+  isSelected?: boolean;
+  /** Flag indicating this option is in the chosen pane. */
+  isChosen?: boolean;
+  /** Flag indicating this option is expanded by default. */
+  defaultExpanded?: boolean;
+  /** Flag indicating this option has a checkbox */
+  hasCheck?: boolean;
```

Comment:
Shouldn't all the tree items have checks?  If so, this should not be needed.

Reply:
I thought to leave it in just in case a user wanted to make something un-checkable (like a folder), but I can remove it.

Reply:
@mmenestr @mcarrano thoughts on this?  Do we want to allow them to not have the checkbox?

Reply:
I don't think so. It's not something I've seen. What do you think @mmenestr?

Reply:
I'll remove it for now then.